eli5/sklearn/permutation_importance.py

```diff
@@ -0,0 +1,252 @@
+            - "prefit" string constant (default).
+
+        If "prefit" is passed, it is assumed that ``estimator`` has been
+        fitted already and all data is used for computing feature importances.
+
+    refit : bool
+        Whether to fit the estimator on the whole data if cross-validation
+        is used (default is False).
+
+    Attributes
+    ----------
+    feature_importances_ : array
+        Feature importances, computed as mean decrease of the score when
+        a feature is permuted (i.e. becomes noise).
+
+    feature_importances_std_ : array
+        Standard deviations of feature importances.
+
+    results_ : list of arrays
+        A list of score decreases for all experiments.
+
+    estimator_ : an estimator
+        The base estimator from which the transformer is built.
+        This is stored only when a non-fitted estimator is passed to the
+        ``PermutationImportance``, i.e when prefit is False.
+
+    rng_ : numpy.random.RandomState
+        random state
+    """
+    def __init__(self, estimator, scoring=None, n_iter=5, random_state=None,
+                 cv='prefit', refit=True):
+        # type: (...) -> None
+        if isinstance(cv, str) and cv != "prefit":
+            raise ValueError("Invalid cv value: {!r}".format(cv))
+        self.refit = refit
+        self.estimator = estimator
+        self.scoring = scoring
+        self.n_iter = n_iter
+        self.random_state = random_state
+        self.cv = cv
+        self.rng_ = check_random_state(random_state)
+
+    def fit(self, X, y, groups=None, **fit_params):
+        # type: (...) -> PermutationImportance
+        """Compute ``feature_importances_`` attribute and optionally
+        fit the base estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The training input samples.
+
+        y : array-like, shape (n_samples,)
+            The target values (integers that correspond to classes in
+            classification, real numbers in regression).
+
+        groups : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+        **fit_params : Other estimator specific parameters
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+
+        if self.cv != "prefit" and self.refit:
+            self.estimator_ = clone(self.estimator)
+            self.estimator_.fit(X, y, **fit_params)
+
+        X = check_array(X)
+
+        if self.cv not in (None, "prefit"):
+            self.results_ = self._cv_feature_importances(
+                X, y, groups=groups, **fit_params)
+        else:
+            self.results_ = self._non_cv_feature_importances(X, y)
+
+        self.feature_importances_ = np.std(self.results_, axis=0)
```

Comment:
wow! 

I had problems making tests to work with np.std, but there is enough correlation between feature importances and their std deviations (more important features often have larger stddev as well), so I was able made it work by choosing right estimators.